frontend/src/views/Crops/CropsModal.tsx

```diff
@@ -332,36 +423,88 @@ function CropsModal({
       setCalculationsPerformed(true);
       return;
     }
-
     if (fieldIndex !== null && selectedCrop !== undefined && selectedCropType !== undefined) {
       try {
-        // Calculate crop requirements (P2O5, K2O, N)
-        const cropRequirementN = getCropRequirementN(formData, selectedCrop, selectedCropType);
-        const cropRequirementP205 = await getCropRequirementP205(
-          formData,
-          field.SoilTest,
-          farmRegion,
-        );
-        const cropRequirementK2O = await getCropRequirementK2O(
-          formData,
-          field.SoilTest,
-          farmRegion,
-        );
-
-        // Calculate crop removals (N, P2O5, K2O)
-        const cropRemovalN = getCropRemovalN(formData, selectedCrop, selectedCropType);
-        const cropRemovalP205 = getCropRemovalP205(formData, selectedCrop, selectedCropType);
-        const cropRemovalK20 = getCropRemovalK20(formData, selectedCrop, selectedCropType);
+        let nutrientValues;
+        // Get soil test P and K values, use defaults if not available
+        const soilTestValP = field.SoilTest?.convertedKelownaP
+          ? Number(field.SoilTest.convertedKelownaP)
+          : 250; // Default from defaultSoilTestData
+        const soilTestValK = field.SoilTest?.convertedKelownaK
+          ? Number(field.SoilTest.convertedKelownaK)
+          : 500; // Default from defaultSoilTestData
+        // Leaf tissue will be completed after this ticket and updated here. Using temp values for now
+        const leafTissueP = 1.1;
+        const leafTissueK = 1.1;
+        if (
+          selectedCropType.id === CROP_TYPE_BERRIES_ID &&
+          selectedCrop.id === CROP_RASPBERRIES_ID
```

Comment:
I think you should just be able to check `selectedCrop.id === CROP_RASPBERRIES_ID` and same below for blueberries.